sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -93,6 +94,13 @@ private[hive] class HiveThriftServer2(hiveContext: HiveContext)
   extends HiveServer2
   with ReflectedCompositeService {
 
+  private[hive] val uiTab: Option[ThriftServerTab] =
+    if (hiveContext.hiveconf.getBoolean("spark.ui.enabled", true)) {
```

Comment:
It's a bit weird to read Spark configuration via `HiveConf` although it's valid here. `hiveContext.sparkContext.getConf` might be better.